### ISSUE TITLE
Kernel panic causing sudden shutdown of the machine

### ISSUE THREAD

**Issue by SHuang-Broad** [AdguardTeam/AdguardForMac#866]
Log to be sent to Apple

```
panic(cpu 0 caller 0xffffff800d547aea): "tcp_lock: so=0xffffff8092e779e0 NO PCB! lr=0xffffff7f945aefcf lrh= 0xffffff800d6dc57b:0xffffff800d6fb266 0xffffff800d6caf7d:0xffffff800d6fb266 0xffffff800d6daf1d:0xffffff800d6fb266 0xffffff800d6daf1d:0xffffff800d538422 \n"@/AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/xnu/xnu-6153.141.28.1/bsd/netinet/tcp_subr.c:3309
Backtrace (CPU 0), Frame : Return Address
0xffffffa3f3acba10 : 0xffffff800d11763d 
0xffffffa3f3acba60 : 0xffffff800d251b25 
0xffffffa3f3acbaa0 : 0xffffff800d2436ae 
0xffffffa3f3acbaf0 : 0xffffff800d0bda40 
0xffffffa3f3acbb10 : 0xffffff800d116d07 
0xffffffa3f3acbc10 : 0xffffff800d1170f7 
0xffffffa3f3acbc60 : 0xffffff800d8be82c 
0xffffffa3f3acbcd0 : 0xffffff800d547aea 
0xffffffa3f3acbd00 : 0xffffff800d6cdf08 
0xffffffa3f3acbd20 : 0xffffff7f945aefcf 
0xffffffa3f3acbd70 : 0xffffff800d6fb0c2 
0xffffffa3f3acbda0 : 0xffffff800d6cb13d 
0xffffffa3f3acbdc0 : 0xffffff800d549a58 
0xffffffa3f3acbe30 : 0xffffff800d5496ae 
0xffffffa3f3acbe50 : 0xffffff800d507a41 
0xffffffa3f3acbec0 : 0xffffff800d159605 
0xffffffa3f3acbf40 : 0xffffff800d159131 
0xffffffa3f3acbfa0 : 0xffffff800d0bd13e 
      Kernel Extensions in backtrace:
         com.adguard.nfext(27.0)[4AF16C66-CBFC-369B-8F0A-6FB96F1D536E]@0xffffff7f945ad000->0xffffff7f945d8fff

BSD process name corresponding to current thread: kernel_task
Boot args: chunklist-security-epoch=0 -chunklist-no-rev2-dev

Mac OS version:
19H1030

Kernel version:
Darwin Kernel Version 19.6.0: Mon Apr 12 20:57:45 PDT 2021; root:xnu-6153.141.28.1~1/RELEASE_X86_64
Kernel UUID: BB1171FD-A9E3-3D24-92F7-F4DD45974EFD
Kernel slide:     0x000000000ce00000
Kernel text base: 0xffffff800d000000
__HIB  text base: 0xffffff800cf00000
System model name: MacBookPro16,1 (Mac-E1008331FDC96864)
System shutdown begun: NO

System uptime in nanoseconds: 173178409145042
last loaded kext at 155880585753741: >AudioAUUC	1.70 (addr 0xffffff7f91470000, size 28672)
last unloaded kext at 156260859888087: >!UAudio	323.4 (addr 0xffffff7f945d9000, size 434176)
loaded kexts:
com.adguard.nfext	27
com.cisco.kext.acsock	4.9.40
>AudioAUUC	1.70
>DiskImages.UDIFDiskImage	493.0.0
>DiskImages.FileBackingStore	493.0.0
@filesystems.smbfs	3.4.4
@kext.AMDRadeonX6000	3.1.0
@kext.AMDRadeonServiceManager	3.1.0
>!AGraphicsDevicePolicy	5.2.7
@fileutil	20.036.15
@AGDCPluginDisplayMetrics	5.2.7
>!AHV	1
|IOUserEthernet	1.0.1
|IO!BSerialManager	7.0.6f8
>pmtelemetry	1
>!APlatformEnabler	2.7.0d0
>AGPM	111.4.4
>X86PlatformShim	1.0.0
>!AUpstreamUserClient	3.6.8
@Dont_Steal_Mac_OS_X	7.0.0
>AGDCBacklightControl	5.2.7
>!A!IKBLGraphics	14.0.7
>BridgeAudioCommunication	6.70.7
>!AThunderboltIP	3.1.4
>!ATopCaseHIDEventDriver	3430.1
>!AHIDALSService	1
>!ABridgeAudio!C	6.70.7
>!A!ICFLGraphicsFramebuffer	14.0.7
>!A!ISlowAdaptiveClocking	4.0.0
>!AAVEBridge	6.1
>!AMuxControl2	5.2.7
>!AGFXHDA	100.1.429
>!A!IPCHPMC	2.0.1
>!AMCCSControl	1.14
@filesystems.autofs	3.0
>BCMWLANFirmware4355.Hashstore	1
>BCMWLANFirmware4364.Hashstore	1
>BCMWLANFirmware4377.Hashstore	1
>!ABCMWLANBusInterfacePCIe	1
@filesystems.hfs.kext	522.100.5
@BootCache	40
@!AFSCompression.!AFSCompressionTypeDataless	1.0.0d1
@!AFSCompression.!AFSCompressionTypeZlib	1.0.0
>!AVirtIO	1.0
@filesystems.apfs	1412.141.2
@private.KextAudit	1.0
>!ASmartBatteryManager	161.0.0
>!AACPIButtons	6.1
>!ASMBIOS	2.1
>!AACPIEC	6.1
>!AAPIC	1.7
$!AImage4	1
@nke.applicationfirewall	303
$TMSafetyNet	8
@!ASystemPolicy	2.0.0
|EndpointSecurity	1
>DiskImages.KernelBacked	493.0.0
@kext.AMDRadeonX6100HWLibs	1.0
@kext.AMDRadeonX6000HWServices	3.1.0
|IOAVB!F	850.1
>!AHIDKeyboard	209
|IO!BHost!CUARTTransport	7.0.6f8
|IO!BHost!CTransport	7.0.6f8
@!AGPUWrangler	5.2.7
>!AHS!BDriver	3430.1
>IO!BHIDDriver	7.0.6f8
>!AActuatorDriver	3440.1
>!AMultitouchDriver	3440.1
>!AInputDeviceSupport	3440.8
@kext.AMDRadeonX6000Framebuffer	3.1.0
>!ABacklightExpert	1.1.0
|IOAccelerator!F2	438.7.4
>!A!ILpssUARTv1	3.0.60
>!A!ILpssUARTCommon	3.0.60
>!AOnboardSerial	1.0
|IOSlowAdaptiveClocking!F	1.0.0
>!AGraphicsControl	5.2.7
|IONDRVSupport	576.1
@kext.AMDSupport	3.1.0
@!AGraphicsDeviceControl	5.2.7
>X86PlatformPlugin	1.0.0
>IOPlatformPlugin!F	6.0.0d8
>!ASMBus!C	1.0.18d1
|IOGraphics!F	576.1
@plugin.IOgPTPPlugin	840.3
|IOEthernetAVB!C	1.1.0
@kext.triggers	1.0
>usb.!UHub	1.2
>usb.cdc.ncm	5.0.0
>usb.cdc	5.0.0
>usb.networking	5.0.0
>usb.!UHostCompositeDevice	1.2
>!ABCMWLANCore	1.0.0
>mDNSOffloadUserClient	1.0.1b8
>IOImageLoader	1.0.0
|IO80211!FV2	1200.12.2b1
>corecapture	1.0.4
|IOSkywalk!F	1
|IOSurface	269.11
@filesystems.hfs.encodings.kext	1
|IOAudio!F	300.2
@vecLib.kext	1.2.0
|IOSerial!F	11
>usb.!UVHCIBCE	1.2
>usb.!UVHCI	1.2
>usb.!UVHCICommonBCE	1.0
>usb.!UVHCICommon	1.0
>!AEffaceableNOR	1.0
|IOBufferCopy!C	1.1.0
|IOBufferCopyEngine!F	1
|IONVMe!F	2.1.0
>!AThunderboltPCIDownAdapter	2.5.4
>!AThunderboltDPInAdapter	6.2.6
>!AThunderboltDPAdapter!F	6.2.6
>!AHPM	3.4.4
>!A!ILpssI2C!C	3.0.60
>!A!ILpssDmac	3.0.60
>!A!ILpssI2C	3.0.60
>!AThunderboltNHI	5.8.6
|IOThunderbolt!F	7.6.1
|IOUSB!F	900.4.2
>usb.!UXHCIPCI	1.2
>usb.!UXHCI	1.2
>!AEFINVRAM	2.1
>!AEFIRuntime	2.1
>!ASMCRTC	1.0
|IOSMBus!F	1.1
|IOHID!F	2.0.0
$quarantine	4
$sandbox	300.0
@kext.!AMatch	1.0.0d1
>!AKeyStore	2
>!UTDM	489.120.1
|IOSCSIBlockCommandsDevice	422.120.3
>!ACredentialManager	1.0
>!AFDEKeyStore	28.30
>!AEffaceable!S	1.0
>!AMobileFileIntegrity	1.0.5
@kext.CoreTrust	1
|CoreAnalytics!F	1
|IOTimeSync!F	840.3
|IONetworking!F	3.4
>DiskImages	493.0.0
|IO!B!F	7.0.6f8
|IO!BPacketLogger	7.0.6f8
>!ASSE	1.0
>KernelRelayHost	1
>!ASEPManager	1.0.1
>IOSlaveProcessor	1
|IOUSBMass!SDriver	157.140.1
|IOSCSIArchitectureModel!F	422.120.3
|IO!S!F	2.1
|IOUSBHost!F	1.2
>usb.!UCommon	1.0
>!UHostMergeProperties	1.2
>!ABusPower!C	1.0
|IOReport!F	47
>!AACPIPlatform	6.1
>!ASMC	3.1.9
>watchdog	1
|IOPCI!F	2.9
|IOACPI!F	1.4
@kec.pthread	1
@kec.corecrypto	1.0
@kec.Libm	1
```

**Comment by SHuang-Broad** [comment]
OK this is getting annoying, 2 days in a row, for a payed user

```
panic(cpu 4 caller 0xffffff800a147aea): "tcp_lock: so=0xffffff8060b593e8 NO PCB! lr=0xffffff7f91095d2c lrh= 0xffffff800a2dc57b:0xffffff800a2fb266 0xffffff800a2caf7d:0xffffff800a2fb266 0xffffff800a2daf1d:0xffffff800a2fb266 0xffffff800a2daf1d:0xffffff800a138422 \n"@/AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/xnu/xnu-6153.141.28.1/bsd/netinet/tcp_subr.c:3309
Backtrace (CPU 4), Frame : Return Address
0xffffffa3eef0ba10 : 0xffffff8009d1763d 
0xffffffa3eef0ba60 : 0xffffff8009e51b25 
0xffffffa3eef0baa0 : 0xffffff8009e436ae 
0xffffffa3eef0baf0 : 0xffffff8009cbda40 
0xffffffa3eef0bb10 : 0xffffff8009d16d07 
0xffffffa3eef0bc10 : 0xffffff8009d170f7 
0xffffffa3eef0bc60 : 0xffffff800a4be82c 
0xffffffa3eef0bcd0 : 0xffffff800a147aea 
0xffffffa3eef0bd00 : 0xffffff800a2cdf08 
0xffffffa3eef0bd20 : 0xffffff7f91095d2c 
0xffffffa3eef0bd70 : 0xffffff800a2fb0c2 
0xffffffa3eef0bda0 : 0xffffff800a2cb13d 
0xffffffa3eef0bdc0 : 0xffffff800a149a58 
0xffffffa3eef0be30 : 0xffffff800a1496ae 
0xffffffa3eef0be50 : 0xffffff800a107a41 
0xffffffa3eef0bec0 : 0xffffff8009d59605 
0xffffffa3eef0bf40 : 0xffffff8009d59131 
0xffffffa3eef0bfa0 : 0xffffff8009cbd13e 
      Kernel Extensions in backtrace:
         com.adguard.nfext(27.0)[4AF16C66-CBFC-369B-8F0A-6FB96F1D536E]@0xffffff7f91093000->0xffffff7f910befff

BSD process name corresponding to current thread: kernel_task
Boot args: chunklist-security-epoch=0 -chunklist-no-rev2-dev chunklist-security-epoch=0 -chunklist-no-rev2-dev

Mac OS version:
19H1030

Kernel version:
Darwin Kernel Version 19.6.0: Mon Apr 12 20:57:45 PDT 2021; root:xnu-6153.141.28.1~1/RELEASE_X86_64
Kernel UUID: BB1171FD-A9E3-3D24-92F7-F4DD45974EFD
Kernel slide:     0x0000000009a00000
Kernel text base: 0xffffff8009c00000
__HIB  text base: 0xffffff8009b00000
System model name: MacBookPro16,1 (Mac-E1008331FDC96864)
System shutdown begun: NO

System uptime in nanoseconds: 34501508331167
last loaded kext at 33931911596488: >usb.IOUSBHostHIDDevice	1.2 (addr 0xffffff7f8dfdf000, size 45056)
last unloaded kext at 34055624197080: >!UAudio	323.4 (addr 0xffffff7f910bf000, size 434176)
loaded kexts:
com.adguard.nfext	27
com.cisco.kext.acsock	4.9.40
@filesystems.smbfs	3.4.4
>AudioAUUC	1.70
@kext.AMDRadeonX6000	3.1.0
@kext.AMDRadeonServiceManager	3.1.0
>!AGraphicsDevicePolicy	5.2.7
@fileutil	20.036.15
@AGDCPluginDisplayMetrics	5.2.7
>!AHV	1
|IOUserEthernet	1.0.1
|IO!BSerialManager	7.0.6f8
>AGPM	111.4.4
>!APlatformEnabler	2.7.0d0
>X86PlatformShim	1.0.0
>pmtelemetry	1
>!AUpstreamUserClient	3.6.8
@Dont_Steal_Mac_OS_X	7.0.0
>AGDCBacklightControl	5.2.7
>!A!IKBLGraphics	14.0.7
>BridgeAudioCommunication	6.70.7
>!AThunderboltIP	3.1.4
>!AHIDALSService	1
>!ABridgeAudio!C	6.70.7
>!A!ISlowAdaptiveClocking	4.0.0
>!AMCCSControl	1.14
>!AAVEBridge	6.1
>!AMuxControl2	5.2.7
>!ATopCaseHIDEventDriver	3430.1
>!AGFXHDA	100.1.429
>!A!ICFLGraphicsFramebuffer	14.0.7
>!A!IPCHPMC	2.0.1
@filesystems.autofs	3.0
>BCMWLANFirmware4355.Hashstore	1
>BCMWLANFirmware4364.Hashstore	1
>BCMWLANFirmware4377.Hashstore	1
>!ABCMWLANBusInterfacePCIe	1
@filesystems.hfs.kext	522.100.5
@BootCache	40
@!AFSCompression.!AFSCompressionTypeDataless	1.0.0d1
@!AFSCompression.!AFSCompressionTypeZlib	1.0.0
>!AVirtIO	1.0
@filesystems.apfs	1412.141.2
@private.KextAudit	1.0
>!ASmartBatteryManager	161.0.0
>!AACPIButtons	6.1
>!ASMBIOS	2.1
>!AACPIEC	6.1
>!AAPIC	1.7
$!AImage4	1
@nke.applicationfirewall	303
$TMSafetyNet	8
@!ASystemPolicy	2.0.0
|EndpointSecurity	1
@kext.AMDRadeonX6100HWLibs	1.0
@kext.AMDRadeonX6000HWServices	3.1.0
|IOAVB!F	850.1
@!AGPUWrangler	5.2.7
>!ABacklightExpert	1.1.0
@kext.AMDRadeonX6000Framebuffer	3.1.0
>X86PlatformPlugin	1.0.0
|IOSlowAdaptiveClocking!F	1.0.0
>!ASMBus!C	1.0.18d1
>!AGraphicsControl	5.2.7
>!AHIDKeyboard	209
>!AHS!BDriver	3430.1
>IO!BHIDDriver	7.0.6f8
>!AActuatorDriver	3440.1
>!AMultitouchDriver	3440.1
>!AInputDeviceSupport	3440.8
|IONDRVSupport	576.1
|IO!BHost!CUARTTransport	7.0.6f8
|IO!BHost!CTransport	7.0.6f8
>!A!ILpssUARTv1	3.0.60
>!A!ILpssUARTCommon	3.0.60
>!AOnboardSerial	1.0
@kext.AMDSupport	3.1.0
@!AGraphicsDeviceControl	5.2.7
|IOAccelerator!F2	438.7.4
|IOGraphics!F	576.1
>IOPlatformPlugin!F	6.0.0d8
@plugin.IOgPTPPlugin	840.3
|IOEthernetAVB!C	1.1.0
@kext.triggers	1.0
>usb.!UHub	1.2
>usb.cdc.ncm	5.0.0
>usb.cdc	5.0.0
>usb.networking	5.0.0
>usb.!UHostCompositeDevice	1.2
>!ABCMWLANCore	1.0.0
>mDNSOffloadUserClient	1.0.1b8
>IOImageLoader	1.0.0
|IO80211!FV2	1200.12.2b1
>corecapture	1.0.4
|IOSkywalk!F	1
|IOSurface	269.11
@filesystems.hfs.encodings.kext	1
|IOAudio!F	300.2
@vecLib.kext	1.2.0
|IOSerial!F	11
>usb.!UVHCIBCE	1.2
>usb.!UVHCI	1.2
>usb.!UVHCICommonBCE	1.0
>usb.!UVHCICommon	1.0
>!AEffaceableNOR	1.0
|IOBufferCopy!C	1.1.0
|IOBufferCopyEngine!F	1
|IONVMe!F	2.1.0
>!AThunderboltPCIDownAdapter	2.5.4
>!AThunderboltDPInAdapter	6.2.6
>!AThunderboltDPAdapter!F	6.2.6
>!AHPM	3.4.4
>!A!ILpssI2C!C	3.0.60
>!A!ILpssDmac	3.0.60
>!A!ILpssI2C	3.0.60
|IOUSB!F	900.4.2
>!AThunderboltNHI	5.8.6
|IOThunderbolt!F	7.6.1
>usb.!UXHCIPCI	1.2
>usb.!UXHCI	1.2
>!AEFINVRAM	2.1
>!AEFIRuntime	2.1
>!ASMCRTC	1.0
|IOSMBus!F	1.1
|IOHID!F	2.0.0
$quarantine	4
$sandbox	300.0
@kext.!AMatch	1.0.0d1
>!AKeyStore	2
>!UTDM	489.120.1
|IOSCSIBlockCommandsDevice	422.120.3
>!ACredentialManager	1.0
>!AFDEKeyStore	28.30
>!AEffaceable!S	1.0
>!AMobileFileIntegrity	1.0.5
@kext.CoreTrust	1
|CoreAnalytics!F	1
|IOTimeSync!F	840.3
|IONetworking!F	3.4
>DiskImages	493.0.0
|IO!B!F	7.0.6f8
|IO!BPacketLogger	7.0.6f8
>!ASSE	1.0
>KernelRelayHost	1
>!ASEPManager	1.0.1
>IOSlaveProcessor	1
|IOUSBMass!SDriver	157.140.1
|IOSCSIArchitectureModel!F	422.120.3
|IO!S!F	2.1
|IOUSBHost!F	1.2
>usb.!UCommon	1.0
>!UHostMergeProperties	1.2
>!ABusPower!C	1.0
|IOReport!F	47
>!AACPIPlatform	6.1
>!ASMC	3.1.9
>watchdog	1
|IOPCI!F	2.9
|IOACPI!F	1.4
@kec.pthread	1
@kec.corecrypto	1.0
@kec.Libm	1

```

**Comment by Chinaski1** [comment]
Hello there!

Sorry for the late reply. 
I find your request that you sent in [earlier](https://github.com/AdguardTeam/AdguardForMac/issues/391#issuecomment-677960185).
As we noted earlier, unfortunately, we cannot fix the issue on the application side, so I am closing the task.

**Comment by Chinaski1** [comment]
@SHuang-Broad 

Hello again.

It seems I was in a hurry to close this task.
As far as I understand you can cause crash by performing certain actions?
I would like to ask you to describe the steps to reproduce the issue.

**Comment by SHuang-Broad** [comment]
Thanks.

For the last two times, I was simply browsing the internet with Chrome.
I could not recall which sites I was trying to access, so unsure if Adguard detected anything malicious before the sites loaded, but I do recall that the site was quite slow to respond.

I know this is not a lot of information, but the machine did shut down without any warning: went straight to black screen.

Steve

**Comment by scrutinizer11** [comment]
I stumbled across this thread while searching the Internet for my issue, which is exactly as stated by the OP. The panic log is word-for-word identical. "tcp-lock". I've been observing the behaviour for quite a while. It occurs too randomly to note the pattern. Last time it happened, Firefox was running but not the frontmost. I don't remember the details of the previous situations.

I've chimed in without expecting a bugfix, since my AG version is unsupported. Google AI search tips mentioned a special Terminal command to trace conflicting kexts.

```bash
$ kextstat | grep -v apple
Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
   90    0 0xffffff7f80e71000 0x184000   0x184000   at.obdev.nke.LittleSnitch (5474) 21D3E540-82E5-38B3-9030-999B3D754C89 <8 6 5 3 1>
  183    0 0xffffff7f840f6000 0x2c000    0x2c000    com.adguard.nfext (31) 3798A55F-7074-374B-A0BE-0538BD7B4B95 <6 5 1>
```

Both LittleSnitch and AdGuard are compatible with my macOS, and both were dropped. So...
